src/modules/codes/RedeemableCodes.ts

```diff
@@ -200,6 +200,14 @@ export default class RedeemableCodes implements Saveable {
 
                 while (quest.curQuest() < quest.totalQuests) {
                     quest.curQuestObject().complete();
+                    App.game.wallet.gainDungeonTokens(15000);
+                    Notifier.notify({
+                        title: 'Tutorial Skip',
+                        message: 'You have skipped the tutorial, and found a stash of Dungeon Tokens.',
+                        type: NotificationConstants.NotificationOption.warning,
+                        timeout: 1e4,
+                    });
+                    return false;
                 }
```

Comment:
The `gainDungeonTokens` and notification should be after the while loop, not within it. Also don't need the `return false;`

```suggestion
                while (quest.curQuest() < quest.totalQuests) {
                    quest.curQuestObject().complete();
                }
                App.game.wallet.gainDungeonTokens(15000);
                Notifier.notify({
                    title: 'Tutorial Skip',
                    message: 'You have skipped the tutorial, and found a stash of Dungeon Tokens.',
                    type: NotificationConstants.NotificationOption.warning,
                    timeout: 1e4,
                });
```